pkg/spire/spire_gke.go

```diff
@@ -0,0 +1,58 @@
+package spire
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"io/fs"
+	"os"
+)
+
+const SPIFFE_ENDPT = "SPIFFE_ENDPOINT_SOCKET"
+
+func HasSpireEndpoint() bool {
+	s := os.Getenv(SPIFFE_ENDPT)
```

Comment:
```suggestion
	s := os.Getenv(workloadapi.SocketEnv)
```
it's already decleared in the package you're importing

https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/workloadapi